Change Color HTML conversion from ARGB to RGBA

aaronfranke opened the pull request:
Change Color HTML conversion from ARGB to RGBA, add support for 3 and 4 hex digits in C#, add examples of 3 and 4 hex digits in the documentation, and refactor some of the code to be simpler (such as adding a `_parse_col4` method and making `_parse_col8` use it twice). Now it actually matches the HTML/CSS spec everywhere.

~~This conflicts with #40850 since the tests will have to be changed after this PR. It probably would make sense to merge this PR first and rebase/fix the test cases, so that the test cases don't have to be added and then immediately changed here.~~ That was merged, so this PR changes the test cases.

Calinou commented:
I already did the same thing in https://github.com/godotengine/godot/pull/36992, but your PR is more complete as it also adds support for 3-4 hex digits in C#.

aaronfranke committed:
Simplify html_is_valid and allow it to work with 3 and 4 hex digits

aaronfranke commented:
> PS: In a future PR, it would be nice to make `html_is_valid` return `true` for shorthand codes (3-4 hex digits).

I just added this in a 2nd commit (along with greatly simplifying the code - all we need to do is verify the amount of digits and if they are valid hex, not interpret them). Note that this method is only ever used in String's `is_valid_html_color`.

aaronfranke committed:
Fix Color test cases for RGBA

akien-mga commented:
```
15:32 <Akien> reduz: any objection to #40868 ?
15:35 <reduz> Akien: looks good
15:35 <reduz> Akien: but make sure if this function is used somewhere in code, that its fixed. Also if there are settings files using this, they will break
15:35 <reduz> Akien: so I would recommend investigating that if not already
15:36 <Akien> Thanks, I'll ask aaronfranke to have a look at it.
```

aaronfranke committed:
Change Color HTML conversion from ARGB to RGBA

Also add support for 3 and 4 digit values in C#. Now it actually matches the HTML/CSS spec.

aaronfranke commented:
@akien-mga I took a look at everywhere `Color::html` is used and everything seems to be in order. Most of the values in the functions that lead to this method somewhere down the line are either provided by `Color::to_html` or user-provided.

I also looked at every instance of colors being defined with hex codes with alpha via `$ egrep -RIni "#[0-9A-Fa-f]{8}"` and there was exactly one instance where it was still using ARGB (it was just a comment, but anyway I pushed a fix). The vast majority of colors in Godot's codebase are defined in RGB format, without alpha, so those hex codes work the same either way.

akien-mga commented:
Thanks!
